lib/data-types.js

```diff
@@ -352,6 +352,41 @@ BLOB.prototype.toSql = function() {
 };
 
 /**
+ * Range types are data types representing a range of values of some element type (called the range's subtype). Only available in postgres.
+ * See {@link http://www.postgresql.org/docs/9.4/static/rangetypes.html|Postgres documentation} for more details
+ * @property RANGE
+ */
+
+var RANGE = function (Subtype) {
+  var options = {};
+  if(typeof Subtype === 'function') { // if subtype passed - instantiate object of this subtype and return new function
+    options.subtype = new Subtype();
+    return RANGE.bind({}, options);
```

Comment:
No, it won't work if I return new `RANGE` object. In the case you describe, if subtype is specified for `RANGE`, it will return an object. Then, `ARRAY` constructor will check whether it is an object (and it is an object), so `options` variable will be assigned to the `RANGE` object rather than `{ type: type }`. Then, further down it will fail to get `options.type` because it is undefined. Thus, model builder will fail at all with uncaught exception.